src/app/shared/pagination-nav/pagination-nav.component.ts

```diff
@@ -1,29 +1,46 @@
-import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { Component, EventEmitter, OnInit, Output } from '@angular/core';
+import { select, Store } from '@ngrx/store';
+import { AppStateInterface } from '../interfaces/interfaces';
+import { map, Observable } from 'rxjs';
+import {
+  currentPageSelector,
+  maxPageSelector,
+} from '../../store/products-store/products.selectors';
 
 @Component({
   selector: 'app-pagination-nav',
   templateUrl: './pagination-nav.component.html',
 })
 export class PaginationNavComponent implements OnInit {
-  @Input() amountOfPage?: number;
-  @Input() currentPage?: number;
   @Output() pageChanges = new EventEmitter<number>();
-  @Output() onPrevPage = new EventEmitter<void>();
-  @Output() onNextPage = new EventEmitter<void>();
-  pages: number[] = [];
+  @Output() showPrevPage = new EventEmitter<void>();
+  @Output() showNextPage = new EventEmitter<any>();
```

Comment:
Let's add rule to not use ANY. 
Could you please update all ANY type to accurate interfaces?